src/hello_client.erl

```diff
@@ -165,12 +166,16 @@ handle_call(terminate, _From, State) ->
 %% @hidden
 handle_info({?INCOMING_MSG, Message}, State) ->
     incoming_message(Message, State);
+handle_info(?PING, State = #client_state{waiting_for_pong = true, keep_alive_interval = KeepAliveInterval}) ->
+    ?LOG_ERROR("There is no PONG answer on PING for ~p msec. Client will be restarted.", [KeepAliveInterval]),
```

Comment:
There are two things which are kind of irritating here:
1. The client process is restarted and not only the transport handler which is in our case the zmq or http client. The problem which should be fixed here is transport specific.
2.  It would be nice to get to whole time we are waiting for a $PONG since the first $PING where we got no $PONG answer. Currently we only know that a $PING wasn't answered in the KeepAliveInterval. Of course this time can include several restarts of the transport client.